assets/src/components/Sidebar/LegacyThemeSettings.js

```diff
@@ -30,108 +44,111 @@ const themeOptions = [
 
 ];
 
-const ThemeSelect = withPostMeta( SelectControl );
-
-const SelectWithDefaultLabel = compose(
-  fromThemeOptions,
-  withPostMeta,
-  withDefaultLabel,
-)( SelectControl );
-
-const Select = compose(
-  fromThemeOptions,
-  withPostMeta,
-)( SelectControl );
-
-const Radio = compose(
-  fromThemeOptions,
-  withPostMeta,
-)( RadioControl );
-
-const ColorPalette = compose(
-  fromThemeOptions,
-  withPostMeta,
-)( ColorPaletteControl );
-
 export const LegacyThemeSettings = props => {
   const {
     handleThemeSwitch,
     theme,
   } = props;
 
+  const meta = useSelect(select => select('core/editor').getEditedPostAttribute('meta'),[]);
+  const { editPost } = useDispatch('core/editor');
+
+  useEffect(() => {
+    handleThemeSwitch('theme', meta.theme, meta)
```

Comment:
For now I kept these parameters. I'll simplify it when converting `CampaignSidebar` to a function component.